service/frontend/workflow_handler.go

```diff
@@ -2631,6 +2637,27 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
 	}, nil
 }
 
+func (wh *WorkflowHandler) validateScheduledWorkflowSearchAttributes(
+	request *workflowservice.CreateScheduleRequest,
+	namespaceName namespace.Name,
+) error {
+
+	scheduledAction := request.GetSchedule().GetAction().GetAction()
+	if scheduledAction == nil {
+		panic("no scheduled workflow")
+	}
+	action, ok := scheduledAction.(*schedpb.ScheduleAction_StartWorkflow)
+	if !ok {
+		panic(scheduledAction)
+	}
+	searchAttributes := action.StartWorkflow.SearchAttributes
+	unaliased, err := searchattribute.UnaliasFields(wh.saMapperProvider, searchAttributes, namespaceName.String())
+	if err != nil {
+		return err
+	}
+	return wh.validateSearchAttributes(unaliased, namespaceName)
```

Comment:
this is already done here: https://github.com/temporalio/temporal/blob/main/service/frontend/workflow_handler.go#L2679-L2683 inside validateStartWorkflowArgsForSchedule